src/store/modules/auth.js

```diff
@@ -19,24 +19,45 @@ export default {
   },
 
   actions: {
-    signup (context, authData) {
-      authUser(context, authData, 'signUp', URL_KEY)
+    signup ({ commit }, authData) {
+      return axiosAuth.post(`/accounts:signUp?key=${URL_KEY}`, {
+        email: authData.email,
+        password: authData.password,
+        returnSecureToken: true
+      })
+        .then(res => {
+          const tokenId = res.data.idToken
+          document.cookie = `token=${tokenId}; max-age=${res.data.expiresIn}`
+          commit('authUser', {
+            token: tokenId
+          })
+
+          router.push('/')
```

Comment:
Vuex should'nt think about $router here. Router is responsibility of component itself, Vuex - should commit data into store and return promise - that's all. Remove this router.push() and transfer it to component's `then` block.